cmd/main.go

```diff
@@ -106,12 +109,44 @@ func main() {
 	}
 
 	if s.AddIPTablesRule {
+		if err := iptables.ClearRules(); err != nil {
+			log.Fatalf("%s", err)
+		}
+
 		if err := iptables.AddRule(s.AppPort, s.MetadataAddress, s.HostInterface, s.BindIP); err != nil {
 			log.Fatalf("%s", err)
 		}
+		defer func() {
+			err = iptables.ClearChain();
+			if err != nil {
+				log.WithError(err).Errorf("failed to clean iptables chain")
+			}
+		}()
 	}
 
 	if err := s.Run(s.APIServer, s.APIToken, s.NodeName, s.Insecure); err != nil {
 		log.Fatalf("%s", err)
 	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	done := make(chan error)
+	go func() {
+		done <- s.Serve(ctx)
+	}()
+
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGTERM)
+	go func() {
+		sig := <-sigs
+		log.WithField("sig", sig).Info("received signal, cleaning up")
+		cancel()
+		done <- nil
```

Comment:
why do you send a nil to this chan ?
Isn't something done by the listener ?

Reply:
done